### PR TITLE
13939 Overview page

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Ready for review

### CODE REVIEW

File: admin/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -333,3 +333,25 @@ table.spacious-two-column {
   margin-top: 4.4em;
 }
 
+//not sure where to place these yet
```
Comment (GuyHarwood) [inline admin/assets/stylesheets/application.scss]: if unsure about placement speak to Jon

Comment (jon-shipley) [reply admin/assets/stylesheets/application.scss]: May be best to create a new partial just for the sorting

File: admin/views/school/pupil-register.ejs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                </tr>
+            </tbody>
+            <% }) %>
+        </table>
+    </div>
+    <div class="column-one-third sidebar-top-border" tabindex="0" role="group" aria-label="sidebar">
+        <h3 class="heading-medium" tabindex="0">Before check</h3>
+        <ul class="list">
+            <li>
+                <a href="" target="_blank">Group pupils</a>
```
Comment (jon-shipley) [inline admin/views/school/pupil-register.ejs]: Is it in the spec to open every page up in a new window?

File: admin/views/school/school-home.ejs [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
         <ol class="list">
             <li>
-                <h2 class="heading-small"><a href="/school/manage-pupils">Pupil register</a></h2>
+                <h2 class="heading-small"><a href="/school/pupil-register/1">Pupil register</a></h2>
```
Comment (jon-shipley) [inline admin/views/school/school-home.ejs]: Is this `1` a typo perhaps?

File: admin/controllers/school.js [hunk 4]
```diff
@@ -29,6 +29,35 @@ const getHome = async (req, res, next) => {
   })
 }
 
+const getPupils = async (req, res, next) => {
+  res.locals.pageTitle = 'Pupil register'
+  const { sortOrder } = req.params
+  res.locals.sortOrder = sortOrder || 1
+  res.locals.sortClass = parseInt(res.locals.sortOrder) === -1 ? 'triangle down' : 'triangle'
```
Comment (jon-shipley) [inline admin/controllers/school.js]: Should always specify the radix when using parseInt ...`parseInt(var, 10)`

File: admin/controllers/school.js [hunk 5]
```diff
@@ -29,6 +29,35 @@ const getHome = async (req, res, next) => {
   })
 }
 
+const getPupils = async (req, res, next) => {
+  res.locals.pageTitle = 'Pupil register'
+  const { sortOrder } = req.params
+  res.locals.sortOrder = sortOrder || 1
```
Comment (jon-shipley) [inline admin/controllers/school.js]: I think sortOrder needs more validation - what if someone passes in unusual input?

File: admin/services/pupilService.js [hunk 6]
```diff
@@ -53,6 +53,16 @@ const pupilService = {
       score,
       percentage
     }
+  },
+
+  sortRecords: async (sortOrder, schoolId) => {
+    let sortedPupils
+    try {
+      sortedPupils = await Pupil.find({ 'school': schoolId }).sort({ 'lastName': sortOrder }).lean().exec()
```
Comment (jon-shipley) [inline admin/services/pupilService.js]: I like the efficiency of  `lean`. 👍

Comment (GeorgeChatzigiannis) [reply admin/services/pupilService.js]: @jon-shipley will soon post the update not using mongoose as discussed anymore for this

File: admin/views/school/pupil-register.ejs (hunk 2 above)
Comment (GeorgeChatzigiannis) [reply admin/views/school/pupil-register.ejs]: Doesn't seem like, I removed it

**Pull request merged**